package/lib/src/beam_guard.dart

```diff
@@ -20,15 +21,23 @@ class BeamGuard {
     this.replaceCurrentStack = true,
   });
 
-  /// A list of path strings that are to be guarded.
+  /// A list of path strings or regular expressions (using dart's RegExp class) that are to be guarded.
   ///
+  ///For strings:
   /// Asterisk wildcard is supported to denote "anything".
   ///
   /// For example, '/books/*' will match '/books/1', '/books/2/genres', etc.
   /// but will not match '/books'. To match '/books' and everything after it,
   /// use '/books*'.
   ///
   /// See [_hasMatch] for more details.
+  ///
+  /// For RegExp
+  /// You can use RegExp instances and the delegate will check for a match using [RegExp.hasMatch]
+  ///
+  /// For example, `RegExp('/books/')` will match '/books/1', '/books/2/genres', etc.
+  /// but will not match '/books'. To match '/books' and everythin after it,
```

Comment:
Nice PR!👍

I just saw that there is a 'g' missing in the doc comment. It says 'everythin' instead of 'everything'. 😄

Reply:
@hatem-u, meet our meticulous @devj3ns :smiley:

Reply:
😄😂👋

Reply:
LOL Nice catch @devj3ns